packages/babel-plugin-proposal-decorators/src/index.ts

```diff
@@ -37,6 +38,10 @@ export default declare((api, options) => {
   }
 
   if (legacy) {
+    if (version !== undefined) {
+      throw new Error("'version' can't be used with legacy decorators");
+    }
```

Comment:
We could also allow `version: "legacy"`, but throw if both `version` and `legacy` are set. Then in Babel 8 we can remove `legacy: true`, to reduce the number of options.